malaria24/ona/models.py

```diff
@@ -160,11 +165,19 @@ def get_digest_email_data(self):
                         .first().create_date_time.strftime(
                             "%d %B %Y"
                         )
+                    now_start_day = int(start_date.split(' ', 1)[0])
+                    if now_start_day < min_start_day:
+                        min_start_day = now_start_day
+                        start_week_day = start_date
                     end_date = province_cases \
                         .last().create_date_time.strftime(
                             "%d %B %Y"
                         )
-                    week = "{0} to {1}".format(start_date, end_date)
+                    now_end_day = int(end_date.split(' ', 1)[0])
+                    if now_end_day > max_start_day:
+                        max_start_day = now_end_day
+                        end_week_day = end_date
+                    week = "{0} to {1}".format(start_week_day, end_week_day)
```

Comment:
I don't think that this will format the date correctly.

It looks like this will generate something like `7 to 22` for `week`, where previously it would have generated `07 January 2018 to 22 February 2018`

Also, I think for this task we need to keep two start and end dates here. One is for inside the province loop, where we want `week` in line 190 to be the date range for all the cases in the province. The other is for outside the province loop, where we want `week` in line 218 to be the date range across all provinces.